components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -707,6 +707,32 @@ void BraveWalletService::SetSelectedCoin(mojom::CoinType coin) {
   ::brave_wallet::SetSelectedCoin(profile_prefs_, coin);
 }
 
+void BraveWalletService::GetChainIdForActiveOrigin(
+    mojom::CoinType coin,
+    GetChainIdForActiveOriginCallback callback) {
+  GetActiveOrigin(base::BindOnce(
```

Comment:
BraveWalletServiceDelegateImpl::GetActiveOrigin is actually a synchronous functions so this could be refactored without callback. 
But ok to to leave this as is.

Reply:
??? That is always asynchronous, only private`GetActiveOriginInternal` is synchronous.